processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -188,7 +188,7 @@ protected Integer addToFacts(
             lastBuffer.capacity() - bufferOffset >= aggsTotalSize) {
           aggBuffer = lastBuffer;
         } else {
-          ResourceHolder<ByteBuffer> bb = bufferPool.take();
+          ResourceHolder<ByteBuffer> bb = bufferPool.takeOrFailOnTimeout(60000);
```

Comment:
Is this competing with the processing pool? that means you can choke out processing threads on accident while incremental indexing is going on.

Reply:
yes, that would be true.
i think it is more of a limitation of current `OffheapIncrementalIndex` implementation which can not work with fixed amount of resources and would keep on allocating more and more buffers. also, this impl keeps dimensions etc on-heap so doesn't really serve purpose of being off-heap ... things become too slow if dimensions are pushed off-heap due to repeated serde .
FWIW, for above reasons, no-one actually uses current `OffheapIncrementalIndex` implementation and we can possibly remove it.

Reply:
I'm considering the removal of `OffheapIncrementalIndex` in this patch.

Reply:
also it was only ever used by GroupBy-v1 implementation if explicitly configured .. it was not possible to use OffheapIncrementalIndex for indexing actually.